### PR TITLE
feat: enable to preview themes

### PR DESCRIPTION
**Pull request by xiyaowong**
Preview the selected theme, applies it on <cr> or restore original theme
automatically.

### CODE REVIEW

**Comment by Conni2461** [comment]
Cool, works good so far, would also be cool to have the current file opened in the previewer buffer rather than closing the window. Will do code review tomorrow :)

CC @teto

**Comment by xiyaowong** [comment]
yes, that' what I thought at first, but I found it's easier to close than to preview, so I just implement close. I will make some commits later:smile:

File: lua/telescope/builtin/internal.lua [hunk 1]
```diff
@@ -634,26 +634,90 @@ internal.buffers = function(opts)
...
+  if vim.fn.buflisted(bufnr) ~= 1 then
+    local deleted = false
+    local function del_win(win_id)
+      if win_id == nil or not vim.api.nvim_win_is_valid(win_id) then
+        return
+      end
+
+      local bnr = vim.api.nvim_win_get_buf(win_id)
+      if vim.api.nvim_buf_is_valid(bnr) and not vim.api.nvim_buf_get_option(bnr, 'buflisted') then
+        vim.cmd(string.format('silent! bdelete! %s', bnr))
```
Comment (Conni2461) [inline lua/telescope/builtin/internal.lua:655]: We have a util function for that `utils.buf_delete(bufnr)`

https://github.com/nvim-telescope/telescope.nvim/blob/37a3a68a78e15900254ae5d9d4f9f0b5100a1735/lua/telescope/utils.lua#L338-L350

**Comment by tjdevries** [comment]
Only thing I'd add is to add an option to disable auto switching, because some colorschemes do not reload nicely, so it may be best to allow users to simply do the fuzzy finding over them.

Otherwise idea & impl seem pretty good to me

**Comment by xiyaowong** [comment]
got it :)

**Comment by Conni2461** [comment]
LGTM, can we get some docs here https://github.com/nvim-telescope/telescope.nvim/blob/138697980b68ce7d4f0661c752691ffe5dbb72b0/lua/telescope/builtin/init.lua#L255-L257 Like `---@field enable_preview boolean: ....`

I dont think `docgen` will run because its disabled for master (and you are working on master) but i will run it locally then :)

Thanks :)

**Comment by xiyaowong** [comment]
Shall I edit something？Actually, I don't know about docgen 🤣 @Conni2461

**Comment by Conni2461** [comment]
Yeah can you add the documentation there i will generate the vim docs from it ;) basically after 256 before 257 :laughing: If you are still not sure what i am talking about ill do it. Just tell me :laughing:

**Comment by xiyaowong** [comment]
@Conni2461 There is another option called colors here, I think it's useless, and I keep it just for compatibility.

**Comment by Conni2461** [comment]
I am actually not sure what it does :laughing: 

LGTM thanks for the pr :)

**Comment by Conni2461** [comment]
To be more precise, i see what it does but i am not sure of the use case and why it was added. I cant remember

**Comment by teto** [comment]
Just had the opportunity to test, that's dope ! Thanks a lot !

**Pull request merged**